migration-engine/connectors/sql-migration-connector/src/sql_migration.rs

```diff
@@ -52,15 +41,28 @@ pub enum SqlMigrationStep {
     AlterEnum(AlterEnum),
 }
 
+impl Serialize for SqlMigrationStep {
```

Comment:
It's always hard for me to visualize what the serde stuff does. Looks good, but dare to give an example how this looks like? And it's JSON only?

Reply:
There's a new unit test for it at the bottom of the file. I want to remove that impl as soon as we don't need it anymore (i.e. when the old commands are gone). This is a stop-gap impl so the actual content of the migration doesn't need to implement Serialize and doesn't have a public serialized format.

Reply:
What I mean: SqlMigration shouldn't be `Serialize` nor `Deserialize`, it's something private to sql-migration-connector and I don't want external code depend on its shape, like we currently have.